WcaOnRails/app/controllers/api/v0/competitions_controller.rb

```diff
@@ -29,7 +29,34 @@ def schedule
 
   def results
     competition = competition_from_params
-    render json: competition.results
+    render json: competition
+  end
+
+  def event_results
+    competition = competition_from_params
+    event = Event.c_find(params[:event_id])
+    if event
```

Comment:
Ah, and we no longer need this conditional block. We have a separate path for `/results` and `/results/:event_id`, so here we can just do `Event.c_find!(params[:event_id])` to throw exception for invalid id.